garden-backend-service/src/models/user.py

```diff
@@ -1,11 +1,35 @@
+from typing import TYPE_CHECKING, Optional
 from uuid import UUID
 
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy import String
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from src.models._associations import users_saved_gardens
 from src.models.base import Base
 
+if TYPE_CHECKING:
+    from src.models import Garden
+
 
 class User(Base):
     __tablename__ = "users"
     id: Mapped[int] = mapped_column(primary_key=True)
-    username: Mapped[str]
+    username: Mapped[Optional[str]]
     identity_id: Mapped[UUID] = mapped_column(unique=True)
+    name: Mapped[Optional[str]]
+    email: Mapped[Optional[str]] = mapped_column(unique=True)
+    phone_number: Mapped[Optional[str]]
+    skills: Mapped[Optional[list[str]]] = mapped_column(postgresql.ARRAY(String))
+    domains: Mapped[Optional[list[str]]] = mapped_column(postgresql.ARRAY(String))
+    affiliations: Mapped[Optional[list[str]]] = mapped_column(postgresql.ARRAY(String))
+
+    saved_gardens: Mapped[list["Garden"]] = relationship(
+        "Garden",
+        secondary=users_saved_gardens,
+        back_populates="users",
+        lazy="selectin",
+    )
+
+    @property
```

Comment:
I won't pretend to have any experience with using descriptors in sqlalchemy models, but it looks like plain `@property` is not usually the preferred option: https://docs.sqlalchemy.org/en/20/orm/mapped_sql_expr.html

Reply:
on second thought, something like the "last resort" [example](https://docs.sqlalchemy.org/en/20/orm/mapped_sql_expr.html#using-a-plain-descriptor)  using `object_session` to select just the dois might be what we want here. Either that or just building the list of dois in the route itself, which might be simpler to work around with async